runtime/metricsview/executor_rewrite_comparisons_cte.go

```diff
@@ -0,0 +1,94 @@
+package metricsview
+
+import "fmt"
+
+// rewriteComparisonsCTE extracts out the base or comparison query into a CTE depending on the sort field.
+// This is done to enable more efficient query execution. The CTE ise used as FromSelect or JoinComparisonSelect in the AST and also used in corresponding other join query to filter out dimension values that are not present in the base query.
+func (e *Executor) rewriteComparisonsCTE(ast *AST) {
+	if !e.instanceCfg.MetricsComparisonsCTE {
+		return
+	}
+	if ast.Root.JoinComparisonSelect != nil {
+		_ = e.rewriteComparisonsCTENode(ast, ast.Root)
+	}
+}
+
+func (e *Executor) rewriteComparisonsCTENode(a *AST, n *SelectNode) bool {
```

Comment:
Since the logic is quite similar to the approx comparisons here, I wonder if we should just add this rewrite to that one instead of having a separate rewriter?